File: frontend/quiz/src/seleniumTests/Pages/NavBarPage.js
import Page from "./Page";
import { getElementById } from "./Utility";
class NavBarPage extends Page {
  get AboutUsLink() {
    return getElementById(this.driver, "aboutUs");
  }
  get AddQuestionLink() {
    return getElementById(this.driver, "addQuestion");
  }

  async getUsername() {
    const text = await getElementById(this.driver, "user").then(el =>
      el.getText()
    );
    return text;
  }
  get PythonLink() {
    return getElementById(this.driver, "Python");
  }
  get StatisticsLink() {
    return getElementById(this.driver, "stat");
  }
}

export default NavBarPage;

File: backend/selenium/NavBarPage.py
from Utility import getElementById, isVisibleById
from Page import Page
AboutUsLinkId = "aboutUs"
addQuestionlinkId = "addQuestion"
userOnNavBarId = "user"
PythonLinkId = "Python"
statisticsLinkId = "stat"


class NavBarPage(Page):
    def AboutUsLink(self):
        return getElementById(self.driver, AboutUsLinkId)

    def AddQuestionLink(self):
        return getElementById(self.driver, addQuestionlinkId)

    def getUsername(self):
        text = (getElementById(self.driver, userOnNavBarId)).text
        return text

    def PythonLink(self):
        return getElementById(self.driver, PythonLinkId)

    def StatisticsLink(self):
        return getElementById(self.driver, statisticsLinkId)
